testing/sdk_metrics.py

```diff
@@ -53,15 +48,33 @@ def get_metrics(package_name, service_name, task_name):
     if not task_info:
         return []
 
-    container_id = task_info["status"]["containerStatus"]["containerId"]["value"]
+    task_container_id = task_info["status"]["containerStatus"]["containerId"]["value"]
+
+    # Not related to functionality but consuming this
+    # endpoint to verify downstream integrity
+    containers_url = "{}/system/v1/agent/{}/metrics/v0/containers".format(
+        shakedown.dcos_url(), agent_id)
+    containers_response = sdk_cmd.request("GET", containers_url, retry=False)
```

Comment:
looks like `containers_response` is only used for debugging missing container ids. maybe move this retrieval into the failure case at L77? in other words only get/print the list of containers on the agent if the one we expected to be there was missing

Reply:
Wouldn't we only know that the expected container is missing from the list by comparing against the containers list, thereby making the retrieval necessary anyway?

Reply:
I'd assume the flow would look like:
1. test does retrieval against expected container id
2. retrieval fails with 404 or similar error
3. test returns failure: retrieves list of known container ids to be included in failure message

This assumes that the metrics API explicitly returns an error in step 2. However, if that's wrong and dcos-metrics returns e.g. an empty list, then we wouldn't know the difference between "container id isn't known" and "container with that id hasn't started sending metrics yet". In that case then the current behavior would make more sense. Any idea which is the case?

Reply:
Ah, yeah it's expected that the metrics API return an error if it can't handle the given request, in which case we don't distinguish why there's problem and we do a retry from one layer higher. 

In the case the request succeeds but the list is empty, we still fail and do a retry as we expect some container id to be reported.